### DISCUSSION TITLE
Street name widget to retain function during navigation

### DISCUSSION THREAD

**Discussion by sweeoon** [osmandapp/OsmAnd#18974] [Ideas]
I have the street name widget at the top of my phone screen. It shows the trail or street name I'm currently cycling on -  when I'm just using the map and not navigating. 

However, when I am navigating a trail I've imported into osmand, the street name widget stops functioning. Instead it only shows turn arrows, not the name of the trail/street/road I'm currently riding on. 

Can this be fixed please, so the widget continues to function as expected? It would be so useful during navigation to know what street/trail I'm following.

**Comment by DmitryAlexei** [comment]
Please consider using the *Attach to the roads* option. For more information see https://osmand.net/docs/user/navigation/setup/gpx-navigation#attach-to-roads
